nifi-external/nifi-kafka-connect/nifi-kafka-connector/src/main/java/org/apache/nifi/kafka/connect/StatelessNiFiSinkConfig.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.kafka.connect;
+
+import org.apache.kafka.common.config.ConfigDef;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class StatelessNiFiSinkConfig extends StatelessNiFiCommonConfig {
+    public static final String INPUT_PORT_NAME = "input.port";
+    public static final String FAILURE_PORTS = "failure.ports";
+    public static final String HEADERS_AS_ATTRIBUTES_REGEX = "headers.as.attributes.regex";
+    public static final String HEADER_ATTRIBUTE_NAME_PREFIX = "attribute.prefix";
+    protected static final ConfigDef CONFIG_DEF = createConfigDef();
+
+    public StatelessNiFiSinkConfig(Map<?, ?> originals) {
+        super(CONFIG_DEF, originals);
+    }
+
+    protected StatelessNiFiSinkConfig(ConfigDef definition, Map<?, ?> originals) {
+        super(definition, originals);
+    }
+
+    /**
+     * @return The input port name to use when feeding the flow. Can be null, which means the single available input port will be used.
+     */
+    public String getInputPortName() {
+        return getString(INPUT_PORT_NAME);
+    }
+
+    /**
+     * @return The output ports to handle as failure ports. Flow files sent to this port will cause the Connector to retry.
+     */
+    public Set<String> getFailurePorts() {
+        List<String> configuredPorts = getList(FAILURE_PORTS);
+        if (configuredPorts == null) {
+            return Collections.emptySet();
+        }
+        return new HashSet<>(configuredPorts);
```

Comment:
Recommend using `LinkedHashSet` to preserve ordering.
```suggestion
        return new LinkedHashSet<>(configuredPorts);
```